app/presenters/reservations/calendar_presenter.rb

```diff
@@ -49,6 +14,53 @@ def calendar_object_default
       }
     end
 
+    def self.build(reservation)
+      if reservation.order.present?
+        OrderCalendarPresenter.new(reservation)
+      elsif reservation.offline?
+        OfflineCalendarPresenter.new(reservation)
+      else
+        FallbackCalendarPresenter.new(reservation)
+      end
+    end
+  end
+
+  class OfflineCalendarPresenter < CalendarPresenter
+    def as_calendar_object(_options = {})
+      super.merge(
+        title: "Instrument Offline",
+        email: created_by.try(:full_name),
+      )
+    end
   end
 
+  class OrderCalendarPresenter < CalendarPresenter
+    def as_calendar_object(options = {})
+      ret = super
+
+      return (ret = super) if options[:with_details].blank?
```

Comment:
Can we just return `ret`?

Reply:
yeap, i guess i tried to avoid the method's first line but then forgot about it 🐟